x-pack/plugin/data-streams/src/main/java/org/elasticsearch/xpack/datastreams/DataStreamIndexSettingsProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.datastreams;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.IndexMode;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.shard.IndexSettingProvider;
+
+import java.time.Instant;
+import java.util.Locale;
+import java.util.Optional;
+
+public class DataStreamIndexSettingsProvider implements IndexSettingProvider {
+
+    @Override
+    public Settings getAdditionalIndexSettings(
+        String indexName,
+        String dataStreamName,
+        boolean newDataStream,
+        long resolvedAt,
+        Settings templateAndRequestSettings
+    ) {
+        if (dataStreamName != null && newDataStream) {
+            IndexMode indexMode = Optional.ofNullable(templateAndRequestSettings.get(IndexSettings.MODE.getKey()))
+                .map(value -> IndexMode.valueOf(value.toUpperCase(Locale.ROOT)))
+                .orElse(IndexMode.STANDARD);
+            TimeValue lookAheadTime = templateAndRequestSettings.getAsTime(
+                IndexSettings.LOOK_AHEAD_TIME.getKey(),
+                IndexSettings.LOOK_AHEAD_TIME.getDefault(templateAndRequestSettings)
+            );
+            if (indexMode == IndexMode.TIME_SERIES) {
+                // 0 is the default and this is a required setting and b/c of that another value should be used than 0
+                Instant start = Instant.ofEpochMilli(1);
```

Comment:
Yes, I think a setting like that makes sense. I will update the code to always use `now() - LOOK_AHEAD_TIME` (which can later be the default for this new setting)

Reply:
I've implemented this via: 0c5a7e907d277472eee2539b2450f6d4d858e904
But as you can see in this commit, thus complicates the yaml testing a bit, 
since the test docs need the timestamp to be within the boundary. Not a big issue.

Alternatively we can consider moving tsdb data stream tests to a java rest integration test.
Another change is already needed to make the added assertions work: #81774